docs/model_access_control.md

```diff
@@ -0,0 +1,652 @@
+| `model_access_mode`     | String    | The access mode for this model. Valid values are `public`, `private`, and `restricted`. When this parameter is set to `restricted`, user must specify either `backend_roles` or `add_all_backend_roles`, but not both. Optional. Default is `restricted`. |
+| `backend_roles`         | Array     | A list of the model owner's backend roles to add to the model. Can be specified only if the `model_access_mode` is `restricted`. Cannot be specified at the same time as `add_all_backend_roles`. Optional.                                               |
+| `add_all_backend_roles` | Boolean   | If `true`, all backend roles of the model owner are added to the model group. Default is `false`. Cannot be specified at the same time as `backend_roles`. Admin users cannot set this parameter to `true`. Optional.                                     |
+
+### Response fields
+
+The following table lists the available request fields.
+
+| Field            | Data type | Description                                                      |
+|:-----------------|:----------|:-----------------------------------------------------------------|
+| `model_group_id` | String    | The model group ID that user can use to access this model group. |
+| `status`         | String    | The operation status.                                            |
+
+Any user who has appropriate permissions can create a model group on model group index. Lets say User1 wants to create one. User has three options for the model group — he/she can make it public, private, or restricted (attach the back-end roles HR or IT or both).
+- To create a public Model Group: User should set `model_access_mode` field to `public`
+Any user in the cluster can access this model group and its model versions.
+
+```
+POST /_plugins/_ml/model_groups/_register
+{
+    "name": "test_model_group_public",
+    "description": "This is a public model group",
+    "model_access_mode": "public"
+}
+```
+
+```
+# Response
+{
+    "model_group_id": "GDNmQ4gBYW0Qyy5ZcBcg",
+    "status": "CREATED"
+}
+```
+
+- To create a restrcited Model Group: User should set `model_access_mode` field to `restricted`
+
+When registering a restricted model group, user must attach one or more of their backend roles to the model group using one but not both of the following methods:
+1. Provide a list of backend roles in the `backend_roles` parameter. 
+2. Set the `add_all_backend_roles` parameter to `true` to add all the backend roles of the user to the model group. This option is not available to admin users.
+
+Any user who shares one of the model group's backend roles can access any model in this model group.
+
+An admin user can access all model groups regardless of their access mode.
+
+```
+POST /_plugins/_ml/model_groups/_register
+{
+    "name": "model_group_test",
+    "description": "This is an example description",
+    "model_access_mode": "restricted",
+    "backend_roles" : ["IT"]
+}
+```
+
```

Comment:
May be add a comment here: "Or we can specify `add_all_backend_roles` instead of `backend_roles`?